cpp/include/raft/util/reduction.cuh

```diff
@@ -157,11 +157,11 @@ DI std::pair<T, i_t> blockRankedReduce(T val,
     val = values[lane];
     idx = indices[lane];
   } else {
-    // get the min if it is a max op, get the max if it is a min op
-    val = reduce_op(std::numeric_limits<T>::min(), std::numeric_limits<T>::max()) ==
-              std::numeric_limits<T>::min()
+    // get the lowest if it is a max op, get the max if it is a lowest op
+    val = reduce_op(std::numeric_limits<T>::lowest(), std::numeric_limits<T>::max()) ==
```

Comment:
These are using the largest magnitude positive/negative _finite_ values as the identity of the operation. However, the proper identity of max/min over the floating point domain is negative/positive infinity.

See how cudf handles this here: https://github.com/rapidsai/cudf/blob/85f41dfdb606664cd8bb256dff943bf81aa8816d/cpp/include/cudf/detail/utilities/device_operators.cuh#L134

You might want to check `std::numeric_limits<T>::has_infinity` and use infinities instead of max/lowest.

Also if this needs to support CUDA specific types like __int128 or half-precision, etc, then use cuda::std:: functions from libcudacxx instead of std:: functions.

Reply:
We use these helpers at some other places:
https://github.com/rapidsai/raft/blob/56c0b3a2d3f886ec541e70ac964a60033762d5c0/cpp/include/raft/util/cudart_utils.hpp#L374-L387

Reply:
Makes sense, thanks Bradley! I will do the changes.

Reply:
Great Tamas, no need to reimplement.